<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Scaling coverage data | Introduction to SIPmg</title>
  <meta name="description" content="<p>SIPmg is a statistical analysis package as part of a stable isotope probing
(SIP) metagenomics study to identify isotope incorporating MAGs.The package
utilizes either the qSIP model, ΔBD method, or DeSeq2-based methods such as
HR-SIP and MW-HR-SIP based on the HTSSIP package. For more information on HTSSIP
R package, please visit their vignette</p>" />
  <meta name="generator" content="bookdown 0.31 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Scaling coverage data | Introduction to SIPmg" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/EMBERR_logo-2.png" />
  <meta property="og:description" content="<p>SIPmg is a statistical analysis package as part of a stable isotope probing
(SIP) metagenomics study to identify isotope incorporating MAGs.The package
utilizes either the qSIP model, ΔBD method, or DeSeq2-based methods such as
HR-SIP and MW-HR-SIP based on the HTSSIP package. For more information on HTSSIP
R package, please visit their vignette</p>" />
  <meta name="github-repo" content="ZielsLab/SIPmg.github.io" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Scaling coverage data | Introduction to SIPmg" />
  
  <meta name="twitter:description" content="<p>SIPmg is a statistical analysis package as part of a stable isotope probing
(SIP) metagenomics study to identify isotope incorporating MAGs.The package
utilizes either the qSIP model, ΔBD method, or DeSeq2-based methods such as
HR-SIP and MW-HR-SIP based on the HTSSIP package. For more information on HTSSIP
R package, please visit their vignette</p>" />
  <meta name="twitter:image" content="/images/EMBERR_logo-2.png" />

<meta name="author" content="Pranav Sampara, Kate Waring, Ryan Ziels" />


<meta name="date" content="2023-01-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="quick-example.html"/>
<link rel="next" href="identifying-incorporators.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="background.html"><a href="background.html#stable-isotope-probing"><i class="fa fa-check"></i><b>2.1</b> Stable isotope probing</a></li>
<li class="chapter" data-level="2.2" data-path="background.html"><a href="background.html#quantitative-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Quantitative metagenomics</a></li>
<li class="chapter" data-level="2.3" data-path="background.html"><a href="background.html#quantitative-sip-qsip"><i class="fa fa-check"></i><b>2.3</b> Quantitative SIP (qSIP)</a></li>
<li class="chapter" data-level="2.4" data-path="background.html"><a href="background.html#for-details-on-sip-quantitative-metagenomics-and-qsip-please-refer-to-the-following-works"><i class="fa fa-check"></i><b>2.4</b> For details on SIP, quantitative metagenomics, and qSIP, please refer to the following works:</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="background.html"><a href="background.html#sip"><i class="fa fa-check"></i><b>2.4.1</b> SIP</a></li>
<li class="chapter" data-level="2.4.2" data-path="background.html"><a href="background.html#quantitative-metagenomics-1"><i class="fa fa-check"></i><b>2.4.2</b> Quantitative metagenomics</a></li>
<li class="chapter" data-level="2.4.3" data-path="background.html"><a href="background.html#qsip"><i class="fa fa-check"></i><b>2.4.3</b> qSIP</a></li>
<li class="chapter" data-level="2.4.4" data-path="background.html"><a href="background.html#htssip-r-package"><i class="fa fa-check"></i><b>2.4.4</b> HTSSIP R package</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quick-example.html"><a href="quick-example.html"><i class="fa fa-check"></i><b>3</b> Using SIPmg R package</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quick-example.html"><a href="quick-example.html#example-workflow"><i class="fa fa-check"></i><b>3.1</b> Example workflow</a></li>
<li class="chapter" data-level="3.2" data-path="quick-example.html"><a href="quick-example.html#load-required-data"><i class="fa fa-check"></i><b>3.2</b> Load required data</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="quick-example.html"><a href="quick-example.html#mag-coverage-data"><i class="fa fa-check"></i><b>3.2.1</b> MAG coverage data</a></li>
<li class="chapter" data-level="3.2.2" data-path="quick-example.html"><a href="quick-example.html#sequin-metadata"><i class="fa fa-check"></i><b>3.2.2</b> Sequin metadata</a></li>
<li class="chapter" data-level="3.2.3" data-path="quick-example.html"><a href="quick-example.html#dilutions-data"><i class="fa fa-check"></i><b>3.2.3</b> Dilutions data</a></li>
<li class="chapter" data-level="3.2.4" data-path="quick-example.html"><a href="quick-example.html#fractions-metadata"><i class="fa fa-check"></i><b>3.2.4</b> Fractions metadata</a></li>
<li class="chapter" data-level="3.2.5" data-path="quick-example.html"><a href="quick-example.html#gtdb-style-taxonomy-data"><i class="fa fa-check"></i><b>3.2.5</b> GTDB style taxonomy data</a></li>
<li class="chapter" data-level="3.2.6" data-path="quick-example.html"><a href="quick-example.html#mag-absolute-concentrations"><i class="fa fa-check"></i><b>3.2.6</b> MAG absolute concentrations</a></li>
<li class="chapter" data-level="3.2.7" data-path="quick-example.html"><a href="quick-example.html#gc-content"><i class="fa fa-check"></i><b>3.2.7</b> GC content</a></li>
<li class="chapter" data-level="3.2.8" data-path="quick-example.html"><a href="quick-example.html#log-scale-boolean"><i class="fa fa-check"></i><b>3.2.8</b> Log scale BOOLEAN</a></li>
<li class="chapter" data-level="3.2.9" data-path="quick-example.html"><a href="quick-example.html#coefficient-of-variation"><i class="fa fa-check"></i><b>3.2.9</b> Coefficient of variation</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="quick-example.html"><a href="quick-example.html#coverage-normalization-and-scaling"><i class="fa fa-check"></i><b>3.3</b> Coverage normalization and scaling</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="quick-example.html"><a href="quick-example.html#this-function-uses-the-following-datasets"><i class="fa fa-check"></i><b>3.3.1</b> This function uses the following datasets:</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="quick-example.html"><a href="quick-example.html#preparation-of-scaled-data-for-qsip-analysis"><i class="fa fa-check"></i><b>3.4</b> Preparation of scaled data for qSIP analysis</a></li>
<li class="chapter" data-level="3.5" data-path="quick-example.html"><a href="quick-example.html#estimation-of-atom-fraction-excess"><i class="fa fa-check"></i><b>3.5</b> Estimation of atom fraction excess</a></li>
<li class="chapter" data-level="3.6" data-path="quick-example.html"><a href="quick-example.html#plot-the-atom-fraction-excess-plot"><i class="fa fa-check"></i><b>3.6</b> Plot the atom fraction excess plot</a></li>
<li class="chapter" data-level="3.7" data-path="quick-example.html"><a href="quick-example.html#check-incorporator-list"><i class="fa fa-check"></i><b>3.7</b> Check incorporator list</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html"><i class="fa fa-check"></i><b>4</b> Scaling coverage data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#basic-approach-for-scaling"><i class="fa fa-check"></i><b>4.1</b> Basic approach for scaling</a></li>
<li class="chapter" data-level="4.2" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#choice-of-regression-model"><i class="fa fa-check"></i><b>4.2</b> Choice of regression model</a></li>
<li class="chapter" data-level="4.3" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#scaling-the-data"><i class="fa fa-check"></i><b>4.3</b> Scaling the data</a></li>
<li class="chapter" data-level="4.4" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#comparison-of-the-fits"><i class="fa fa-check"></i><b>4.4</b> Comparison of the fits</a></li>
<li class="chapter" data-level="4.5" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#what-if-there-were-no-sequins-added-in-the-study"><i class="fa fa-check"></i><b>4.5</b> What if there were no sequins added in the study</a></li>
<li class="chapter" data-level="4.6" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#decision-on-the-method-to-scale-coverages"><i class="fa fa-check"></i><b>4.6</b> Decision on the method to scale coverages</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="identifying-incorporators.html"><a href="identifying-incorporators.html"><i class="fa fa-check"></i><b>5</b> SIP metagenomics and identification of isotope incorporators</a>
<ul>
<li class="chapter" data-level="5.1" data-path="identifying-incorporators.html"><a href="identifying-incorporators.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>5.1</b> Comparison of the methods</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="identifying-incorporators.html"><a href="identifying-incorporators.html#afe-method"><i class="fa fa-check"></i><b>5.1.1</b> AFE method</a></li>
<li class="chapter" data-level="5.1.2" data-path="identifying-incorporators.html"><a href="identifying-incorporators.html#mw-hr-sip"><i class="fa fa-check"></i><b>5.1.2</b> MW-HR-SIP</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to SIPmg</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="scaling-coverage-data" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Scaling coverage data<a href="scaling-coverage-data.html#scaling-coverage-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this example we will look at the two scaling approaches that can be utilized in the SIPmg package. The user can decide either</p>
<ol style="list-style-type: decimal">
<li>A robust linear regression model</li>
<li>An ordinary least squares linear regression model
We will also briefly discuss data filtering methods when there are heavily influencing outliers, as a result of upstream methods.</li>
</ol>
<div id="basic-approach-for-scaling" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Basic approach for scaling<a href="scaling-coverage-data.html#basic-approach-for-scaling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Sequencing effort on each fraction and subsequently obtained coverage is typically used to estimate relative abundance for a feature of interest. However, as we discussed in the introductory vignette, this data is compositional and a quantitative estimate of concentrations can be highly informative. For this purpose, synthetic spike-in standards (sequins) with known concentrations are added before recovery of fractionated DNA, from the CsCl medium into TE buffer or water. For details on sequins please refer to <a href="https://www.nature.com/articles/s41467-018-05555-0">Hardwick et al. (2018)</a>. These reference standards in each fraction, can now be used to estimate the concentration of the feature of interest in each fraction.</p>
<p>The approach used for scaling is detailed in the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>For each fraction, sequin coverages and known concentrations are obtained to make a standard curve.</p></li>
<li><p>Limit of detection of sequins, i.e., the lowest concentration where at least one sequin has detectable coverage, is determined for each group of sequin concentrations.</p></li>
<li><p>For each group of sequin concentrations, mean, standard deviation, and the coefficient of variation of coverage are determined. Groups of sequins with coefficient of variation greater than the set threshold will be flagged.</p></li>
<li><p>Sequin groups with coverage values above the limit of detection and below the coefficient of variation threshold are filtered in preparation for regression.</p></li>
<li><p>Linear regression or robust linear regression, based on the user choice, is performed on sequin coverage values as the independent variable and concentration of the sequin group as the dependent variable. The user can also decide on performing log scaling of coverage and concentration values for regression.</p></li>
<li><p>The regression parameters are extracted and plots with regression parameters and best fit line are saved for inspection.</p></li>
<li><p>Coverage values above the limit of detection and below the coefficient of variation threshold are filtered for further analysis. The rest of the values are flagged and reported.</p></li>
<li><p>The filtered values are scaled based on regression parameters to estimate absolute concentrations in each library.</p></li>
<li><p>Absolute concentrations of biological features are saved as a fresh dataset for the subsequent isotope incorporator identification and AFE estimation pipeline.</p></li>
</ol>
</div>
<div id="choice-of-regression-model" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Choice of regression model<a href="scaling-coverage-data.html#choice-of-regression-model" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Ordinary least squares regression can be very sensitive to outliers, resulting a poor model performance when the data is contaminated with outliers. Although there are methods, like Cook’s distance based filtering to address outliers, it may not always be the best idea to remove these data points.</p>
<p>See <a href="https://statisticsbyjim.com/basics/remove-outliers/">these</a> <a href="https://stats.stackexchange.com/questions/175/how-should-outliers-be-dealt-with-in-linear-regression-analysis">discussions</a> and more for better insights into handling outliers.</p>
<p>Robust linear regression is an alternative to this situation. Robust linear regression assigns appropriate weights to the data points, minimizing the influence of outliers on the model performance, without deleting data. In this pipeline, a Huber loss function from the <a href="https://cran.r-project.org/web/packages/MASS/MASS.pdf"><code>MASS</code> R package</a> was used for robust linear regression.</p>
<p>The user is free to choose between the two methods for regression.</p>
</div>
<div id="scaling-the-data" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Scaling the data<a href="scaling-coverage-data.html#scaling-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For illustrating the differences in the regression models, a different dataset, that was impacted by one or many upstream methods, will be used than the one in the introductory vignette.</p>
<p>This dataset can be found here</p>
<p>For both scaling functions the following data and parameters are required</p>
<ol style="list-style-type: decimal">
<li><p>Coverage data. The output of <a href="https://github.com/Ecogenomics/CheckM/wiki/Utility-Commands#coverage">checkm coverage</a> command can be directly used.</p></li>
<li><p>Sequin dilutions and metadata.</p></li>
<li><p>Whether or not log10 scaling to be performed (a BOOLEAN value of TRUE or FALSE).</p></li>
<li><p>Coefficient of variation. In this dataset, a higher coefficient of variation compared to the value used in the example workflow of the <a href="quick-example.html#quick-example">introduction vignette</a> is used to allow data scaling to occur.</p></li>
</ol>
<p>If the robust linear regression is chosen, the function is <code>scale_features_rlm</code> which is performed the function file <code>sequin_scaling_rlm.R</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="scaling-coverage-data.html#cb13-1" aria-hidden="true" tabindex="-1"></a>mag_tab_scaled_rlm <span class="ot">&lt;-</span> SIPmg<span class="sc">::</span><span class="fu">scale_features_rlm</span>(f_tibble, sequins, seq_dil, log_scale, <span class="at">coe_of_variation =</span> coe_of_variation, <span class="at">save_plots =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>If a linear regression is chosen, the function is <code>scale_features_lm</code> which is performed by the function file <code>sequin_scaling_lm.R</code>.</p>
<p>If the user chooses linear regression, they can choose to perform outlier filtering based on the traditional Cook’s distance threshold of 4/n (n being the sample size). Only the outliers in the sequin coverage data are flagged as outliers with the Cook’s distance threshold and are filtered out. Later, the remaining data is subjected to linear regression to obtain model parameters and to scale coverage values of feature of interest.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="scaling-coverage-data.html#cb14-1" aria-hidden="true" tabindex="-1"></a>mag_tab_scaled_lm <span class="ot">&lt;-</span> SIPmg<span class="sc">::</span><span class="fu">scale_features_lm</span>(f_tibble, sequins, seq_dil, log_scale, <span class="at">coe_of_variation =</span> coe_of_variation, <span class="at">cook_filtering =</span> <span class="cn">TRUE</span>, <span class="at">save_plots =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="comparison-of-the-fits" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Comparison of the fits<a href="scaling-coverage-data.html#comparison-of-the-fits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Robust linear regression provides a better model fit compared to the linear regression. Additionally, robust linear regression has less variability compared to the linear regression model.</p>
<p>As discussed previously, removing the outliers may not always be the best idea. However, the outliers can be visualized to assess how “far out” are the outliers and how influential the data points are, negatively influencing the linear regression method.</p>
<p>The <code>mag_tab_scaled_lm</code> function provides the plots for the visualization of Cook’s distance for the sequin data in each fraction. In the plot below, an outlier is visualized. For this data, the Cook’s distance threshold was 0.09, and the outlier had a Cook’s distance of 10.8. The highly influential data point could well be the reason the linear regression model had a poor fit.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="scaling-coverage-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Cook&#39;s distance threshold of the data set</span></span>
<span id="cb15-2"><a href="scaling-coverage-data.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="dv">4</span><span class="sc">/</span>(<span class="fu">length</span>(mag_tab_scaled_lm<span class="sc">$</span>scale_fac<span class="sc">$</span>cooksd[[<span class="dv">3</span>]]))</span></code></pre></div>
<pre><code>## [1] 0.1142857</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="scaling-coverage-data.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Cook&#39;s distance of the outlier before filtering the data</span></span>
<span id="cb17-2"><a href="scaling-coverage-data.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(mag_tab_scaled_lm<span class="sc">$</span>scale_fac<span class="sc">$</span>cooksd[[<span class="dv">3</span>]])</span></code></pre></div>
<pre><code>## [1] 0.3588489</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="scaling-coverage-data.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Before filtration</span></span>
<span id="cb19-2"><a href="scaling-coverage-data.html#cb19-2" aria-hidden="true" tabindex="-1"></a>cooksd_example <span class="ot">=</span> knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;cooksd-example.png&quot;</span>)</span></code></pre></div>
<p>Upon filtering the data to remove the outlier, the fit becomes much better for the linear regression model</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="scaling-coverage-data.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Linear regression plot with filtered outliers in sequin data</span></span>
<span id="cb20-2"><a href="scaling-coverage-data.html#cb20-2" aria-hidden="true" tabindex="-1"></a>filtered_lm_example <span class="ot">=</span> knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;filtered_lm-example.png&quot;</span>)</span></code></pre></div>
</div>
<div id="what-if-there-were-no-sequins-added-in-the-study" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> What if there were no sequins added in the study<a href="scaling-coverage-data.html#what-if-there-were-no-sequins-added-in-the-study" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In our study, we realized that the sequin-scaled data provided the best correlation (Spearman correlation coefficient = 0.85), compared to methods without the use of sequins. However, not all studies would have access to sequins or could choose not to add sequins. We tested qSIP methods with absolute concentration of MAGs obtained as a product of fraction DNA concentration and either with MAG relative abundance (as performed by <a href="https://journals.asm.org/doi/10.1128/mSystems.00151-20">Greenlon et al.</a>) or relative coverage (as performed by <a href="https://journals.asm.org/doi/10.1128/mSphere.00085-21">Starr et al.</a>), or simply with relative coverage, i.e., without converting the MAG coverages to absolute concentration. We found that without sequins, simply using relative coverage provided better specificity (0.99) and good correlation (Spearman correlation coefficient = 0.76), compared to the other two methods. The user could choose any of the three approaches to obtain normalized coverages using the equation <code>coverage_normalization()</code> where the parameter approach would be chosen accordingly (either “greenlon”, “starr”, or “relative_coverage”). In this example, simple relative coverage will be estimated without converting to absolute concentration based on fraction DNA concentrations. The relative coverage table can then be converted to a phyloseq object to run incorporator identification and/or AFE estimation. For more details, please refer to the <a href="identifying-incorporators.html#identifying-incorporators">incorporator identification section</a></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="scaling-coverage-data.html#cb21-1" aria-hidden="true" tabindex="-1"></a>f_tibble <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">&quot;mock_input_data/coverage_metadata.csv&quot;</span>)</span>
<span id="cb21-2"><a href="scaling-coverage-data.html#cb21-2" aria-hidden="true" tabindex="-1"></a>rel.cov <span class="ot">=</span> SIPmg<span class="sc">::</span><span class="fu">coverage_normalization</span>(<span class="at">f_tibble =</span> f_tibble, <span class="at">approach =</span> <span class="st">&quot;relative_coverage&quot;</span>)</span>
<span id="cb21-3"><a href="scaling-coverage-data.html#cb21-3" aria-hidden="true" tabindex="-1"></a>mag.table <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(<span class="fu">as.matrix</span>(rel.cov <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;Feature&quot;</span>)), <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>) <span class="co">#Phyloseq OTU table</span></span></code></pre></div>
</div>
<div id="decision-on-the-method-to-scale-coverages" class="section level2 hasAnchor" number="4.6">
<h2><span class="header-section-number">4.6</span> Decision on the method to scale coverages<a href="scaling-coverage-data.html#decision-on-the-method-to-scale-coverages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It can be seen from the above plots that the regression parameters differ based on the method of regression and whether or not data filtering is used. These regression parameters directly influence the estimation of concentrations of features of interest in the microbiome. Thus, the method for regression must be a careful choice.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="scaling-coverage-data.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.2.2 (2022-10-31 ucrt)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19044)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_Canada.utf8  LC_CTYPE=English_Canada.utf8   
## [3] LC_MONETARY=English_Canada.utf8 LC_NUMERIC=C                   
## [5] LC_TIME=English_Canada.utf8    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] SIPmg_1.4           ggpubr_0.5.0        HTSSIP_1.4.1       
##  [4] phyloseq_1.42.0     forcats_0.5.2       stringr_1.5.0      
##  [7] dplyr_1.0.10        purrr_0.3.5         readr_2.1.3        
## [10] tidyr_1.2.1         tibble_3.1.8        ggplot2_3.4.0      
## [13] tidyverse_1.3.2     BiocManager_1.30.19
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.4.1                backports_1.4.1            
##   [3] plyr_1.8.8                  igraph_1.3.5               
##   [5] lazyeval_0.2.2              splines_4.2.2              
##   [7] BiocParallel_1.32.4         GenomeInfoDb_1.34.4        
##   [9] digest_0.6.30               foreach_1.5.2              
##  [11] htmltools_0.5.4             fansi_1.0.3                
##  [13] magrittr_2.0.3              memoise_2.0.1              
##  [15] googlesheets4_1.0.1         cluster_2.1.4              
##  [17] tzdb_0.3.0                  Biostrings_2.66.0          
##  [19] annotate_1.76.0             modelr_0.1.10              
##  [21] matrixStats_0.63.0          vroom_1.6.0                
##  [23] timechange_0.1.1            colorspace_2.0-3           
##  [25] blob_1.2.3                  rvest_1.0.3                
##  [27] haven_2.5.1                 xfun_0.35                  
##  [29] crayon_1.5.2                RCurl_1.98-1.9             
##  [31] jsonlite_1.8.4              survival_3.4-0             
##  [33] iterators_1.0.14            ape_5.6-2                  
##  [35] glue_1.6.2                  gtable_0.3.1               
##  [37] gargle_1.2.1                zlibbioc_1.44.0            
##  [39] XVector_0.38.0              DelayedArray_0.24.0        
##  [41] car_3.1-1                   Rhdf5lib_1.20.0            
##  [43] BiocGenerics_0.44.0         abind_1.4-5                
##  [45] scales_1.2.1                DBI_1.1.3                  
##  [47] rstatix_0.7.1               Rcpp_1.0.9                 
##  [49] xtable_1.8-4                bit_4.0.5                  
##  [51] stats4_4.2.2                httr_1.4.4                 
##  [53] RColorBrewer_1.1-3          ellipsis_0.3.2             
##  [55] farver_2.1.1                pkgconfig_2.0.3            
##  [57] XML_3.99-0.13               sass_0.4.4                 
##  [59] dbplyr_2.2.1                locfit_1.5-9.6             
##  [61] utf8_1.2.2                  labeling_0.4.2             
##  [63] polynom_1.4-1               tidyselect_1.2.0           
##  [65] rlang_1.0.6                 reshape2_1.4.4             
##  [67] AnnotationDbi_1.60.0        munsell_0.5.0              
##  [69] cellranger_1.1.0            tools_4.2.2                
##  [71] cachem_1.0.6                cli_3.4.1                  
##  [73] generics_0.1.3              RSQLite_2.2.19             
##  [75] ade4_1.7-20                 broom_1.0.1                
##  [77] evaluate_0.19               biomformat_1.26.0          
##  [79] fastmap_1.1.0               yaml_2.3.6                 
##  [81] knitr_1.41                  bit64_4.0.5                
##  [83] fs_1.5.2                    KEGGREST_1.38.0            
##  [85] nlme_3.1-160                xml2_1.3.3                 
##  [87] compiler_4.2.2              rstudioapi_0.14            
##  [89] png_0.1-8                   ggsignif_0.6.4             
##  [91] reprex_2.0.2                geneplotter_1.76.0         
##  [93] bslib_0.4.2                 stringi_1.7.8              
##  [95] highr_0.10                  lattice_0.20-45            
##  [97] Matrix_1.5-3                vegan_2.6-4                
##  [99] permute_0.9-7               multtest_2.54.0            
## [101] vctrs_0.5.1                 pillar_1.8.1               
## [103] lifecycle_1.0.3             rhdf5filters_1.10.0        
## [105] jquerylib_0.1.4             data.table_1.14.6          
## [107] bitops_1.0-7                GenomicRanges_1.50.1       
## [109] R6_2.5.1                    bookdown_0.31              
## [111] IRanges_2.32.0              codetools_0.2-18           
## [113] MASS_7.3-58.1               assertthat_0.2.1           
## [115] rhdf5_2.42.0                SummarizedExperiment_1.28.0
## [117] DESeq2_1.38.1               withr_2.5.0                
## [119] S4Vectors_0.36.1            GenomeInfoDbData_1.2.9     
## [121] mgcv_1.8-41                 parallel_4.2.2             
## [123] hms_1.1.2                   grid_4.2.2                 
## [125] rmarkdown_2.19              MatrixGenerics_1.10.0      
## [127] carData_3.0-5               googledrive_2.0.0          
## [129] Biobase_2.58.0              lubridate_1.9.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="quick-example.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="identifying-incorporators.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
